hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -117,6 +117,11 @@ object HoodieSparkSqlWriter {
     }
 
     val jsc = new JavaSparkContext(sparkContext)
+    if (asyncCompactionTriggerFn.isDefined) {
+      if (jsc.getConf.getOption(DataSourceWriteOptions.SPARK_SCHEDULER_ALLOCATION_FILE_KEY).isDefined) {
+        jsc.setLocalProperty("spark.scheduler.pool", DataSourceWriteOptions.SPARK_DATASOURCE_WRITER_POOL_NAME)
```

Comment:
these 2 does not look like hoodie specific settings while `DataSourceWriteOptions` should only have `hoodie.*` settings

Reply:
I agree. but do you have a good suggestion as to where to move this? I could not find a right class to move this only. 
Only option I can think of is to introduce a new class just to hold this 2 new variables.